src/Stryker.Core/Stryker.Core/Mutators/StringMutator.cs

```diff
@@ -12,7 +14,12 @@ public class StringMutator : MutatorBase<LiteralExpressionSyntax>, IMutator
         public override IEnumerable<Mutation> ApplyMutations(LiteralExpressionSyntax node)
         {
             var kind = node.Kind();
-            if (kind == SyntaxKind.StringLiteralExpression && !(node.Parent is ConstantPatternSyntax))
+            // Get objectCreationSyntax to check if it contains a regex type.
+            var root = node.Parent?.Parent?.Parent;
```

Comment:
@richardwerkman Is there no better way to do this?

Reply:
I'm afraid not. A constructor always contains a argumentlist which contains a first argument which contains the string literal. So we always have to look 3 nodes up to see if the literalexpression is in a regex constructor. We also can't loop upwards as it will be less performant and could be error prone.